File: VSOMapView.m
//
//  VSOMapView.m
//  GPS Body Paint
//
//  Created by François Lamboley on 7/28/09.
//  Copyright 2009 VSO-Software. All rights reserved.
//

#import "VSOMapView.h"

#import "VSOUtils.h"

@implementation VSOMapView

- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
{
	[(id <VSOMapViewDelegate>)self.delegate mapViewDidReceiveTouch:self];
	
	return [super hitTest:point withEvent:event];
}

- (void)dealloc
{
	NSDLog(@"Deallocing a VSOMapView");
	
	[super dealloc];
}

@end

File: GPS Body Paint/Views & Controllers/Play View/VSOMapView.swift
/*
 * MapView.swift
 * GPS Body Paint
 *
 * Created by François Lamboley on 2018/11/4.
 * Copyright © 2018 Frost Land. All rights reserved.
 */

import Foundation
import MapKit



/**
Custom MapViewDelegate

Adds a delegate methode to the default MKMapView:

    func mapView(didReceiveTouch mapView: MKMapView) */
@objc
protocol VSOMapViewDelegate : MKMapViewDelegate {
	
	@objc optional func mapView(didReceiveTouch mapView: MKMapView)
	
}

class VSOMapView : MKMapView {
	
//	deinit {
//		NSLog("Deallocing a VSOMapView")
//	}
	
	override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
		(delegate as? VSOMapViewDelegate)?.mapView?(didReceiveTouch: self)
		
		return super.hitTest(point, with: event)
	}
	
}
